src/components/BlogList/BlogList.js

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { fetchData } from '../../services/Api-service';
+import Card from '../Card/Card';
+import './BlogList.css';
+import Search from '../Search/Search';
+import { handleImageError } from '../../utils/common-utils';
+import { FALLBACK_IMAGE } from '../../constants/common-constants';
+import { useDispatch, useSelector } from 'react-redux';
+import { setBlogsData, setSearchTerm, setSelectedBlog, setImageLoaded, setShowNewBlogModal, updateSelectedBlog, setEditMode } from '../../store/blogSlice';
+import NewBlogModal from '../NewBlog/NewBlog';
+import AlertDialog from '../Dialog/AlertDialog';
+import { getItemFromLocalStorage } from '../../utils/local-storage-utils';
+import { BUTTONS, IMAGE_LOADING, NO_BLOGS_FOUND, WARNING_MODAL } from '../../constants/blog-constants';
+
+
+const BlogList = () => {
+    //Dispatch
+    const dispatch = useDispatch();
+    const blogsData = useSelector((state) => state.blog.blogsData);
+    const searchTerm = useSelector((state) => state.blog.searchTerm);
+    const selectedBlog = useSelector((state) => state.blog.selectedBlog);
+    const imageLoaded = useSelector((state) => state.blog.imageLoaded);
+    const showNewBlogModal = useSelector((state) => state.blog.showNewBlogModal);
+    const darkMode = useSelector((state) => state.theme.darkMode);
+    const selectedFilters = useSelector((state) => state.blog.selectedFilters);
+    const editMode = useSelector((state) => state.blog.editMode);
+
+    //States
+    const [showWarningModal, setShowWarningModal] = useState(false);
+    const [clickedBlog, setClickedBlog] = useState(null);
+
+    const titleInputRef = useRef(null);
+
+   
+    /**
+     * Fetch the data from API
+     */
+    useEffect(() => {
+      const fetchDataFromApi = async () => {
+        try {
+          const storedBlogs = JSON.parse(getItemFromLocalStorage('blogs'));
+          if(storedBlogs?.length){
+            dispatch(setBlogsData(storedBlogs));
+            dispatch(setSelectedBlog(storedBlogs[0]));
+          } else {
+            const data = await fetchData('resources/blogs.json');
```

Comment:
Is there any other way that you could think of to fetch data from an external API using redux?

Reply:
I wasn't able to post the data to the API due to permission issues, and so I had the data set in the local.

Reply:
We don't need to post back the data, instead we need push into the local state array and manipulate it.

Reply:
Got it! Sure